### PR TITLE
Cost time optimization for Loudness Zwicker

### PR DESCRIPTION
**Pull request by ErnestoAvedillo**
With this pull request, I modify several files with the objective to improve the speed of the loudness calculation.
In general terms, instead of calculating with values, I propose to work with arrays or vectors when not possible with arrays.
Some of the changes have modified the complete routine and have been quite hard to develop. 
They are not even intuitive so will require some clarification from my side. I would propose to document this change in your folder documentation with an explanation of those routines.
All tests have passed in my version, so they should pass in your system too but would be recommended to be reviewed from your side.
I also would like your collaborators to verify the results with the audios they have, when possible, and if they don't match, please send some feedback to correct the possible discrepancies.

### CODE REVIEW

**Comment by mglesser** [comment]
Hello Ernesto, thanks a lot for this contribution. I will try to review it soon. I will let you know.

**Comment by ErnestoAvedillo** [comment]
Hello Martin
Thanks for your feedback.

This exercise has been a challenge for me and I am excited to know how it
works by you.

I compared both versions of the code with one audio file, and while the
current needs 350 seconds of running time, the one I generated took 7
seconds only.

I am worried I did something wrong because I can't believe this difference.

Hope hearing you soon.
Thanks for all.
Best regards.

Ernesto

**Review by mglesser — CHANGES_REQUESTED** [review]
Hi Ernesto, thanks a lot for this contribution. I've runned the tests and validations successfully. I've just suggested a few minor modifications before the merge. Don't hesitate to contact me if you want to discuss them.

**Comment by mglesser** [comment]
Your implementation even solved an issue we had for one of the test signals from the ISO ! That's wonderfull !

File: tests/sharpness/test_sharpness.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def test_sharpness():
     # Compute sharpness
     sharpness = comp_sharpness(True, sig, fs, method="din")
     S = sharpness["values"]
-
+    print ("S_Value is ", str(S))
```
[the remark this answers is not in the document]
Comment (ErnestoAvedillo) [reply tests/sharpness/test_sharpness.py:44]: Ups sorry I shouldn't have committed this change.

File: validations/loudness_zwicker/validation_loudness_zwicker_stationary.py [hunk 2]
```diff
@@ -176,7 +176,10 @@ def check_compliance(loudness, iso_ref, out_dir):
     # Extract mosqito calculated values
     N = loudness["values"]
     N_specific = loudness["specific values"]
-
+    print ("My N is ",str(N), " N_iso_ref is ", str(N_iso))
```
[the remark this answers is not in the document]
Comment (ErnestoAvedillo) [reply validations/loudness_zwicker/validation_loudness_zwicker_stationary.py:179]: Same as before.
Is my mistake.

File: mosqito/functions/loudness_zwicker/loudness_zwicker_shared.py [hunk 3]
```diff
@@ -34,17 +34,6 @@ def calc_main_loudness(spec_third, field_type):
     nm :  numpy.ndarray
         Core loudness
     """
-    # Ref. ISO 532-1:2017 paragraph A.3
```
[the remark this answers is not in the document]
Comment (ErnestoAvedillo) [reply mosqito/functions/loudness_zwicker/loudness_zwicker_shared.py:37]: I don't remember having deleted this test.
Could have been a mistake.

**Comment by ErnestoAvedillo** [comment]
I am proud to hear this.
Thank you very much.

**Pull request merged**